### DISCUSSION TITLE
Esp32-WROVER Unable to include the header file `esp32-pinctrl.h`

### DISCUSSION THREAD

**Discussion by r0n9Y** [zephyrproject-rtos/zephyr#51862] [Q&A]
**Description:**
I was trying to build for esp32 WROVER and getting the following error:
```
zephyr/boards/xtensa/esp_wrover_kit/esp_wrover_kit-pinctrl.dtsi:8:10: fatal error: dt-bindings/pinctrl/esp32-pinctrl.h: No such file or directory
    8 | #include <dt-bindings/pinctrl/esp32-pinctrl.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
I checked the repository 3.2.0, in both `zephyr/boards/xtensa/esp_wrover_kit/esp_wrover_kit.dts` and `zephyr\dts\xtensa\espressif\esp32.dtsi`. The header file `esp32-pinctrl.h` is included. But in the folder `zephyr\include\zephyr\dt-bindings\pinctrl`. The header file does not exist

 - Windows 10 Pro version 21H2
 - I tried to use `#include <dt-bindings/pinctrl/esp-pinctrl-common.h>` instead of using the missing header but getting the following error
 ```
CMake Error at C:/ncs/v2.1.1/zephyr/cmake/modules/dts.cmake:230 (message):
  gen_defines.py failed with return code: 1
```

**Comment by sylvioalves** [comment]
@r0n9Y I am not able no reproduce the issue. Have you run `west update` as well? Which sample code you have been using for that?

**Comment by timkruse** [comment]
My app is configured as a freestanding app for the `esp32` board and I just ran into the same error message. I'm sure there are several causes but for me the issue was using the environment variable `ZEPHYR_MODULES` to add a custom module instead of `EXTRA_ZEPHYR_MODULES` which made zephyr/cmake not load the correct HAL, which is also module. Reading the [docs](https://docs.zephyrproject.org/latest/develop/application/index.html#important-build-system-variables) says that `ZEPHYR_MODULES` needs to specify a full list of modules (including the espressif-HAL) but `EXTRA_ZEPHYR_MODULES` adds to the existing/predefined modules.
Make sure to remove the `CMakeCache.txt` file after changing to the correct environment variable and configure again.

**Reply by tomvictor** [reply]
@timkruse Did you manage fix the issue?

**Reply by timkruse** [reply]
Yes, after using `EXTRA_ZEPHYR_MODULES` instead of `ZEPHYR_MODULES` the `esp32-pinctrl.h` file was available again.

If I remember correctly I developed a custom module and wanted to make it available to the zephyr environment. So I added my new module to the `ZEPHYR_MODULES` environment variable. This resulted in the error message of the missing `esp32-pinctrl.h` because `ZEPHYR_MODULES` holds **all** modules and by overriding it with my custom module zephyr is in a misconfigured state. Therefore, I used `EXTRA_ZEPHYR_MODULES` to make my custom module available to the zephyr environment.

**Reply by tomvictor** [reply]
Thanks @timkruse for the response, I am trying to write a BLE firmware for ESP using zephyr, I have added the git-submodules and everything but same issue.  Will try this and update.

**Comment by tonivss** [comment]
I have the same issue as @r0n9Y. The folder/file `dt-bindings/pinctrl/esp32-pinctrl.h` does not exist. I also tried just using the `zephyr/dt-bindings/pinctrl/esp-pinctrl-common.h` (which i also had to modify from `zephyr/dt-bindings/pinctrl/esp32-pinctrl-common.h` to `zephyr/dt-bindings/pinctrl/esp-pinctrl-common.h`). 
The error I get then:
`devicetree error: C:/ncs/v2.6.1/zephyr/boards/xtensa/esp32_devkitc_wrover/esp32_devkitc_wrover-pinctrl.dtsi:15 (column 14): parse error: expected number or parenthesized expression`

Have you been able to solve the problem? If yes, how?
